### DISCUSSION TITLE
Climate sentence not work

### DISCUSSION THREAD

**Discussion by brano6** [OHF-Voice/intents#1875] [_Q&A] [answered]
Hi 
When i ask my HA  What is temperature in living room (climate) in my languge. I get response "An unexpected error during intent recognition"
I try analyze it with developer tools in assist, and there i find what is probbably bad. In protocol is a lot of targets icluding  domain climate,
how could i fix it ? 
thanks
![Snímka obrazovky 2024-01-14 111701](https://github.com/home-assistant/intents/assets/148708429/ade046af-5cf5-4005-84f2-26082f60d38b)
 for help

**Comment by tetele** [comment]
The unexpected error should be in the HA core logs. Could you please paste it here?

**Reply by brano6** [reply]
hello sorry for later response i was on holidays here is HA core log 
Thank for help
[config_entry-hassio-c94c5d429db7577b669d483017e329d7.json (1).txt](https://github.com/home-assistant/intents/files/14015207/config_entry-hassio-c94c5d429db7577b669d483017e329d7.json.1.txt)

**Reply by tetele** [reply]
That's not the core log. You can find the core log here https://my.home-assistant.io/redirect/logs/

**Reply by brano6** [reply]
Hello sorry  now i send issue what occured when i try i ask what is temperature in livingroom it is in notepad
[Unexpected error during intent recognition.txt](https://github.com/home-assistant/intents/files/14042667/Unexpected.error.during.intent.recognition.txt)

**Reply by v1k70rk4** [reply]
Hmm, interesting error. The cause of the error is that instead of a number, you get this response: `Kúrenie` which might be heating? and this cannot be processed by the response template, which expects a number, and therefore it fails.

Response template:
```
    HassClimateGetTemperature:
      default: "{{ state.state }} {{ 'stupeň' if state.state | int | abs == 1 else 'stupne' if state.state | int | abs in [2,3,4] else 'stupňov' }}"
```

**Reply by v1k70rk4** [reply, marked as answer]
Ah, I've figured it out...
The response template is incorrect in Slovak. The correct one should be:
```
        {% set temperature = state_attr(state.entity_id, 'current_temperature') %}
        {{ temperature }} °C
```

But the Slovak response queries the {{ state.state }} status.
Thus, "Kúrenie" would be correct, but since it's not a number, it results in an error.

If you have the opportunity, please correct it like this:
```
        {% set temperature = state_attr(state.entity_id, 'current_temperature') %}
        {{ temperature }} {{ 'stupeň' if temperature | int | abs == 1 else 'stupne' if temperature | int | abs in [2,3,4] else 'stupňov' }}
```

**Reply by brano6** [reply]
Hello thank s for help now it is working for me. It should be good to repair it in official there on github but a i dont have acces to do that. 
And i have another problem with set climate when i say in my language. "Set temp in living room on 20" issue ocured, Unknown Intent HassClimateSetTemperature
![Snímka obrazovky 2024-01-27 152431](https://github.com/home-assistant/intents/assets/148708429/cf5d7fed-ded1-4fb2-856d-ece1dca00733)
![Snímka obrazovky 2024-01-27 152624](https://github.com/home-assistant/intents/assets/148708429/fda51609-30ae-4727-8021-ec3ab34eec58)
 I adding my system log. Thanks a lot

**Reply by v1k70rk4** [reply]
Is 'obývačke' set as an alias for the living room ('obývačka')? I have no other immediate guess. Because it runs fine in the tests, but for the tests, the living room also has 4 aliases (obývačk(a|e|u|y)). Another possibility could be whether there's anything to control there, but I'm leaning more towards the former error.

Regarding the other error, please open a bug ticket and your Slovak language lead will fix it.
https://github.com/home-assistant/intents/issues

Also, when reporting a bug, don't insert a screenshot, as it was a joy to rewrite your question :D. It's better to paste the log formatted in CODE form, and make sure to describe the language, the question asked, and the answer received.

**Reply by brano6** [reply]
Yes have set an alias for living room my alias working for example with lights. Need i download and put file fixtures from this path 
tests/sk/fixtures.yaml ? or its included in HA. I want control my TRV s.

**Reply by v1k70rk4** [reply]
Well, let's see what matches the question you asked in your case. I couldn't manage to adjust the temperature in Slovak because the recognition went to the wrong template (the templates are very similar).

For example:
```
question: Nastav teplotu v nappali na 20 stupňov
sentence_template: <set> <name> na {color}
```

```
question: Prestav teplotu v nappali na 21
sentence_template: <set> <name> na <brightness>
```

```
question: Prestav teplotu na 21
sentence_template:  <set> <name> na <brightness>
```
It might be more accurate to specify what is being adjusted for color and brightness.
 e.g.,
```
Nastav farbu <name> na červenú <v obývačke>.
Nastav jas <name> na 20% <v obývačke>.
```

Please paste in a full log (not an image here) in code format.

**Reply by brano6** [reply]
Hello I try ask: " Nastav teplotu v obývačke na 20 stupňov" And i get this to my logs
HA log:
[HA.log](https://github.com/home-assistant/intents/files/14171270/HA.log)

Assist log:
[intent_results (3).json](https://github.com/home-assistant/intents/files/14171282/intent_results.3.json)

**Reply by v1k70rk4** [reply]
Sorry, I didn't mention where I'm asking the log from. :) Please send a log from here as well (https://my.home-assistant.io/redirect/developer_assist/).

It seems like there's no match, and I'm trying to figure out what the system is confusing it with.

**Reply by brano6** [reply]
I sended log from developer assist it is in previous comment.

**Reply by v1k70rk4** [reply]
Sorry if that was the case. Which HA Core version are you using? Mine looks quite different. Yours is missing the sentence_template: line.

**Reply by brano6** [reply]
I am using HA Core 2024.2.0

**Reply by v1k70rk4** [reply]
Well, this is where my knowledge stops. :) I don't understand why there is no match. With my system set to Slovak, I see the error. But your log looks completely different to you.

Just to repeat myself, I think the setting clashes with the brightness and color adjustment, but the logs you sent do not support this.
```
question: Nastav teplotu v nappali na 20 stupňov
sentence_template: <set> <name> na {color}
```
```
question: Prestav teplotu v nappali na 21
sentence_template: <set> <name> na <brightness>
```
```
question: Prestav teplotu na 21
sentence_template:  <set> <name> na <brightness>
```
It might be more accurate to specify what is being adjusted for color and brightness.
e.g.,

```
Nastav farbu <name> na červenú <v obývačke>.
Nastav jas <name> na 20% <v obývačke>.
```

**Reply by brano6** [reply]
Now i try check it again in HA assist and now you can see sentence template. But before it is not there. 

HA Assist log:
[HA_assist log.txt](https://github.com/home-assistant/intents/files/14228431/HA_assist.log.txt)

File which is in HA assistl log: file":  "sk/climate_HassClimateGetTemperature.yaml" is not modified.

**Comment by v1k70rk4** [comment]
Well dear @brano6  :) We have thoroughly explored this, once it's enabled, it will work for you....

Based on the logs, everything seems fine, so I took a deeper look to see what the issue is, and this is what I found https://github.com/home-assistant/intents/blob/main/intents.yaml :)

```
HassClimateSetTemperature:
  supported: false
```

The reason we both see it in our own languages as well is because we have a custom folder. But currently, it doesn't work in English either.
